### DISCUSSION TITLE
After the es query hits a match, the number of alarm aggregations is inconsistent

### DISCUSSION THREAD

**Discussion by 1998729** [jertel/elastalert2#1101] [Q&A] [answered]
config:
name: 5xx-base-20230201153656
type: frequency
index: ee-sla-test-base-%Y.%m.%d
use_strftime_index: true
run_every:
  minutes: 1
buffer_time:
  minutes: 3
timeframe:
  minutes: 1
num_events: 10
filter:
- query:
    query_string:
      query: 'business: "base" AND http_host: "abc.com"
        AND whitelist.keyword: 0 NOT uri: websocket NOT uri: ws NOT uri: socket.io
        AND status: 500'
query_key:
- status
- uri
alert:
- alertmanager
alertmanager_hosts:
- http://alertmanager.monitoring.svc.cluster.local:9093
alertmanager_api_version: v2
alertmanager_annotations:
  name: aaaa
  severity: P3
alertmanager_labels:
  __channel__: oncall
  __receiver__: aaaa
alertmanager_alert_text_labelname: message
alert_text_args:
- http_host
- uri
- status
- num_matches
alert_subject_args:
- http_host
- uri
- status
- num_matches
alert_text_type: alert_text_only
alert_text: '{0} {1} {2}: {3}<br/><a href="https://gra/d/-ktSqPOVz/sla-gao-jing-da-pan-elastalert2?orgId=3"></a>'
alertmanager_fields:
  #count: num_matches
  http_code: status
  http_host: http_host
  http_uri: uri

elastalert2 log info: 
from 2023-02-03 11:50 CST to 2023-02-03 11:53 CST: 90 query hits (55 already seen), 4 matches, 2 alerts sent

real render text
abc.com /api/v1/trigger/5xx1 500: 4

**Comment by 1998729** [comment]
The number I set is num_events = 10, but it hits 4 and also alarmed

**Comment by 1998729** [comment]
But what can be confirmed is that the alarm threshold that appears after I change status: >=500 to status: 500 in the es query condition is incorrect.

**Comment by 1998729** [comment]
help thank you ~

**Comment by jertel** [comment, marked as answer]
You'd have to review the entire log. Each query adds the count of matches to the running total. Once the running total exceeds the threshold the alert is triggered. The log you provided only shows the most recent query.
